server/core/src/main/java/dev/slimevr/protocol/rpc/settings/RPCSettingsHandler.kt

```diff
@@ -0,0 +1,368 @@
+
+			if (toggles != null) {
+				// Note: toggles.has____ returns the same as toggles._____ this
+				// seems like a bug
+				hpm.setToggle(SkeletonConfigToggles.EXTENDED_SPINE_MODEL, toggles.extendedSpine())
+				hpm
+					.setToggle(
+						SkeletonConfigToggles.EXTENDED_PELVIS_MODEL,
+						toggles.extendedPelvis(),
+					)
+				hpm.setToggle(SkeletonConfigToggles.EXTENDED_KNEE_MODEL, toggles.extendedKnee())
+				hpm
+					.setToggle(
+						SkeletonConfigToggles.FORCE_ARMS_FROM_HMD,
+						toggles.forceArmsFromHmd(),
+					)
+				hpm.setToggle(SkeletonConfigToggles.EXTENDED_SPINE_MODEL, toggles.extendedSpine())
+				hpm
+					.setToggle(
+						SkeletonConfigToggles.EXTENDED_PELVIS_MODEL,
+						toggles.extendedPelvis(),
+					)
+				hpm.setToggle(SkeletonConfigToggles.EXTENDED_KNEE_MODEL, toggles.extendedKnee())
+				hpm
+					.setToggle(
+						SkeletonConfigToggles.FORCE_ARMS_FROM_HMD,
+						toggles.forceArmsFromHmd(),
+					)
+				hpm.setToggle(SkeletonConfigToggles.FLOOR_CLIP, toggles.floorClip())
+				hpm
+					.setToggle(
+						SkeletonConfigToggles.SKATING_CORRECTION,
+						toggles.skatingCorrection(),
+					)
+				hpm.setToggle(SkeletonConfigToggles.VIVE_EMULATION, toggles.viveEmulation())
+				hpm.setToggle(SkeletonConfigToggles.TOE_SNAP, toggles.toeSnap())
+				hpm.setToggle(SkeletonConfigToggles.FOOT_PLANT, toggles.footPlant())
+				hpm.setToggle(SkeletonConfigToggles.SELF_LOCALIZATION, toggles.selfLocalization())
+			}
+
+			if (ratios != null) {
+				if (ratios.hasImputeWaistFromChestHip()) {
+					hpm
+						.setValue(
+							SkeletonConfigValues.WAIST_FROM_CHEST_HIP_AVERAGING,
+							max(0.0, ratios.imputeWaistFromChestHip().toDouble()).toFloat(),
+						)
+				}
+				if (ratios.hasImputeWaistFromChestLegs()) {
+					hpm
+						.setValue(
+							SkeletonConfigValues.WAIST_FROM_CHEST_LEGS_AVERAGING,
+							max(0.0, ratios.imputeWaistFromChestLegs().toDouble()).toFloat(),
+						)
+				}
+				if (ratios.hasImputeHipFromChestLegs()) {
+					hpm
+						.setValue(
+							SkeletonConfigValues.HIP_FROM_CHEST_LEGS_AVERAGING,
+							max(0.0, ratios.imputeHipFromChestLegs().toDouble()).toFloat(),
+						)
+				}
+				if (ratios.hasImputeHipFromWaistLegs()) {
+					hpm
+						.setValue(
+							SkeletonConfigValues.HIP_FROM_WAIST_LEGS_AVERAGING,
+							max(0.0, ratios.imputeHipFromWaistLegs().toDouble()).toFloat(),
+						)
+				}
+				if (ratios.hasInterpHipLegs()) {
+					hpm
+						.setValue(
+							SkeletonConfigValues.HIP_LEGS_AVERAGING,
+							max(0.0, ratios.interpHipLegs().toDouble()).toFloat(),
+						)
+				}
+				if (ratios.hasInterpKneeTrackerAnkle()) {
+					hpm
+						.setValue(
+							SkeletonConfigValues.KNEE_TRACKER_ANKLE_AVERAGING,
+							max(0.0, ratios.interpKneeTrackerAnkle().toDouble()).toFloat(),
+						)
+				}
+				if (ratios.hasInterpKneeAnkle()) {
+					hpm
+						.setValue(
+							SkeletonConfigValues.KNEE_ANKLE_AVERAGING,
+							max(0.0, ratios.interpKneeAnkle().toDouble()).toFloat(),
```

Comment:
wut is this
pls use float `max` or `coerceAtLeast`
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.ranges/coerce-at-least.html
```suggestion
							max(0f, ratios.imputeWaistFromChestHip()),
						)
				}
				if (ratios.hasImputeWaistFromChestLegs()) {
					hpm
						.setValue(
							SkeletonConfigValues.WAIST_FROM_CHEST_LEGS_AVERAGING,
							max(0f, ratios.imputeWaistFromChestLegs()),
						)
				}
				if (ratios.hasImputeHipFromChestLegs()) {
					hpm
						.setValue(
							SkeletonConfigValues.HIP_FROM_CHEST_LEGS_AVERAGING,
							max(0f, ratios.imputeHipFromChestLegs()),
						)
				}
				if (ratios.hasImputeHipFromWaistLegs()) {
					hpm
						.setValue(
							SkeletonConfigValues.HIP_FROM_WAIST_LEGS_AVERAGING,
							max(0f, ratios.imputeHipFromWaistLegs()),
						)
				}
				if (ratios.hasInterpHipLegs()) {
					hpm
						.setValue(
							SkeletonConfigValues.HIP_LEGS_AVERAGING,
							max(0f, ratios.interpHipLegs()),
						)
				}
				if (ratios.hasInterpKneeTrackerAnkle()) {
					hpm
						.setValue(
							SkeletonConfigValues.KNEE_TRACKER_ANKLE_AVERAGING,
							max(0f, ratios.interpKneeTrackerAnkle()),
						)
				}
				if (ratios.hasInterpKneeAnkle()) {
					hpm
						.setValue(
							SkeletonConfigValues.KNEE_ANKLE_AVERAGING,
							max(0f, ratios.interpKneeAnkle()),
```